clientconn.go

```diff
@@ -435,7 +437,19 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 	if cc.dopts.copts.Dialer == nil {
 		cc.dopts.copts.Dialer = newProxyDialer(
 			func(ctx context.Context, addr string) (net.Conn, error) {
-				return dialContext(ctx, "tcp", addr)
+				network := "tcp"
+				p := regexp.MustCompile(`^[a-z]+://`)
+				if p.MatchString(addr) {
+					t, err := url.Parse(addr)
+					if err != nil {
+						return nil, fmt.Errorf("grpc: %v", err)
+					}
+					if t.Scheme == "unix" && t.Host != "" {
+						network = t.Scheme
+						addr = t.Host
```

Comment:
`url.Parse("unix://addr")` gives me
```go
(*url.URL)(0xc4200b0000)({
  Scheme: (string) (len=4) "unix",
  ...
  Host: (string) (len=4) "addr",
  Path: (string) "",
  ...
})
```

while `url.Parse("unix:///addr")` gives me
```go
(*url.URL)(0xc42009c000)({
  Scheme: (string) (len=4) "unix",
  ...
  Host: (string) "",
  Path: (string) (len=5) "/addr",
  ...
})
```

And `url.Parse("unix:/addr")` gives me
```go
(*url.URL)(0xc4200c2000)({
  Scheme: (string) (len=4) "unix",
  ...
  Host: (string) "",
  Path: (string) (len=5) "/addr",
  ...
})
```

How about dialing to host if path is empty string?
So that `path` takes priority over `host`, and `unix://authority/addr` would also behave correctly.
```go
addr = t.Path
if addr == "" {
  addr = t.Host
}
```